packages/blade/src/tokens/theme/createTheme.ts

```diff
@@ -0,0 +1,379 @@
+const WCAG2ContrastOptions: WCAG2Options = {
+  level: 'AAA',
+  size: 'large',
+};
+
+/**
+ * getColorWithOpacity
+ * @param color - The color to add opacity to
+ * @param opacity - The opacity to add to the color
+ * @returns The color with the opacity added
+ * @example
+ * getColorWithOpacity('#fff', 0.5) // returns 'hsla(0, 0%, 100%, 0.5)'
+ *
+ **/
+const getColorWithOpacity = (
+  color: ColorInput,
+  opacity: number,
+): `hsla(${number}, ${number}%, ${number}%, ${number})` => {
+  const hslColorObj = tinycolor(color).toHsl();
+  return `hsla(${hslColorObj.h}, ${hslColorObj.s * 100}%, ${hslColorObj.l * 100}%, ${opacity})`;
+};
+
+/**
+ *
+ * @param baseColorInput - The base color to generate the chromatic color palette from in hex, rgb, or hsl format
+ * @description
+ * Generates a chromatic color palette based on the base color passed in.
+ * The base color is used to generate a palette of 11 colors, 5 shades lighter and 5 shades darker than the base color.
+ * @returns Array of chromatic color palette
+ */
+const generateChromaticBrandColors = (baseColorInput: ColorInput): Color['chromatic']['azure'] => {
+  const baseColor = tinycolor(baseColorInput);
+  const baseColorHslString = baseColor.toHslString();
+  if (__DEV__) {
+    if (!baseColor.isValid())
+      throwBladeError({
+        message: 'Invalid brandColor passed',
+        moduleName: 'createTheme',
+      });
+  }
+
+  const palette = [baseColorHslString]; // Include the original color
+  const brightness = tinycolor(baseColor).getBrightness();
+  // Determine how much to lighten or darken the colors depending on the brightness of the base color
+  const lightnessFactor = brightness > 150 ? 3 : 6;
+  const darknessFactor = brightness < 50 ? 3 : 5;
+
+  let currentColor = baseColor;
+
+  // Generate shades lighter
+  for (let i = 0; i < 6; i++) {
+    currentColor = currentColor.brighten(lightnessFactor);
+    palette.push(currentColor.toHslString());
+  }
+
+  currentColor = tinycolor(baseColorHslString); // Reset to the base color
+
+  // Generate shades darker
+  for (let i = 0; i < 4; i++) {
+    currentColor = currentColor.darken(darknessFactor);
+    palette.unshift(currentColor.toHslString()); // Add shades at the beginning of the palette
+  }
+
+  const colorPalette = palette.reverse();
+  const brandPrimaryColor = colorPalette[6];
+
+  const brandColors = {
+    '50': colorPalette[0],
+    '100': colorPalette[1],
+    '200': colorPalette[2],
+    '300': colorPalette[3],
+    '400': colorPalette[4],
+    '500': colorPalette[5],
+    '600': brandPrimaryColor,
+    '700': colorPalette[7],
+    '800': colorPalette[8],
+    '900': colorPalette[9],
+    '950': colorPalette[10],
+    a00: getColorWithOpacity(brandPrimaryColor, opacity[0]),
+    a50: getColorWithOpacity(brandPrimaryColor, opacity[1]),
+    a100: getColorWithOpacity(brandPrimaryColor, opacity[2]),
+    a200: getColorWithOpacity(brandPrimaryColor, opacity[3]),
```

Comment:
Are we just determining opacity based on descending order or there is a pattern to it? like we did for normal colors where we have darkness/lightness factor.